### ISSUE TITLE
ODPC: Documenthandelingen publiceren

### ISSUE THREAD

**Issue by felixcicatt** [GPP-Woo/GPP-app#26]
Als medewerker wil ik bij ieder document documenthandelingen kunnen vastleggen, zodat de burger dit kan raadplegen.

Let op! [ODRC #76](https://github.com/GeneriekPublicatiePlatformWoo/registratie-component/issues/76) is randvoorwaardelijk!

**Acceptatie criteria**

- Bij ieder bestaand document kan ik de documenthandeling zien.
  - Let op dat het scherm overzichtelijk blijft!
- Wanneer ik een document nieuw toevoeg aan een publicatie, dan moet ik een documenthandeling toevoegen,
  - Er wordt een foutmelding getoond wanneer bij het opslaan niet aan alle documenten een documenthandeling is gekoppeld.
- Wanneer ik een publicatie wijzig, dan kan ik bij ieder document de documenthandeling wijzigen, 
- Wanneer ik een documenthandeling toevoeg of wijzig, dan... : 
  - .... moet ik een keuze maken in de soort handeling: "ondertekening", "ontvangst" of "vaststelling".
    - ! Deze waardelijst is [landelijk ](https://[standaarden.overheid.nl](https://standaarden.overheid.nl/tooi/waardelijsten/work?work_uri=https%3A%2F%2Fidentifier.overheid.nl%2Ftooi%2Fset%2Fccw_plooi_documenthandelingen)/tooi/waardelijsten/work?work_uri=https%3A%2F%2Fidentifier.overheid.nl%2Ftooi%2Fset%2Fccw_plooi_documenthandelingen) bepaald. Stem even af met Maykin of dat jullie deze lijst hardcoded in het ODPC moeten inbakken of dat zij deze lijst via een API moeten ontsluiten.
    - ! In de waardelijst staat ook een hoofd-handeling "PLOOI dopcumenthandeling" (Code: k_d381b512). Deze geven we niet als optie aan gebruikers.
  - ... moet ik een "Datum handeling" opgeven.
  - .... krijg ik bij het opslaan een foutmelding als ik het niet de verplichte velden heb ingevuld..
- Documenthandelingen worden opgehaald uit en weggeschreven in het ODRC via de API.
    - In het ODRC zit bij een documenthandeling ook een attribuut `wasAssociatedWith`. Dit optionele veld gebruiken we in het ODPC nog even niet.

**Taken**

- [ ] 

### === v v v OUD v v v ===
In het formulier waarmee je kan publiceren moet een mogelijkheid komen documenthandelingen toe te voegen. Er zijn drie soorten documenthandeling:

- ondertekening
- ontvangst
- veststelling

In de wireframe is een opzet gedaan voor hoe dit er in het formulier uit zou kunnen zien, maar dit moet nog gecheckt worden.

_Documenthandelingen zijn samengesteld uit een datum, een handeling en een voor de handeling verantwoordelijke organisatie. Aan actieve openbaarmaking in termen van de WOO ligt altijd minimaal één documenthandeling ten grondslag, daarom is dit veld verplicht; meerdere documenthandelingen kunnen worden toegekend aan één document._

Foutafhandeling: als er een waardelijst niet kan worden opgehaald of het formulier kan niet worden opgeslagen, komt er een generieke foutmelding in het scherm. Als het formulier niet kan worden opgeslagen, kan je het opnieuw proberen.

Als je sessie verlopen is, krijg je de optie om opnieuw in te loggen en het opnieuw te proberen (zonder dat je werk verliest).

Documentatie van alle configuratie / environment variabelen met voorbeelden. In eerste instantie in een installation.md bestand in de ODPC github repository.

Zie scherm `Publiceren` in [de wireframe](https://app.uizard.io/p/8dd0f9fa/fullscreen)

[Zie deze comment voor de structuur van een publicatie](https://github.com/GeneriekPublicatiePlatformWoo/registratie-component/issues/2#issuecomment-2326140879)

**Comment by felixcicatt** [comment]
Het is me nog niet helemaal duidelijk hoe een ambtenaar tot een of meer documenthandelingen komt bij een publicatie. Goed om nog te overleggen

**Comment by nijmra** [comment]
@MarcoKlerks Hoe komen we aan de lijst van documenthandelingen? Is dat ook een waardeljist uit ODRC?

**Comment by MarcoKlerks** [comment]
@nijmra 
In de standaard is het een waardelijst: https://[standaarden.overheid.nl](https://standaarden.overheid.nl/tooi/waardelijsten/work?work_uri=https%3A%2F%2Fidentifier.overheid.nl%2Ftooi%2Fset%2Fccw_plooi_documenthandelingen)/tooi/waardelijsten/work?work_uri=https%3A%2F%2Fidentifier.overheid.nl%2Ftooi%2Fset%2Fccw_plooi_documenthandelingen

Maar omdat het een kort lijstje is, wilde @sergei-maertens  het volgens mij hard-coded inbakken. Of was dat @felixcicatt ?... Owww, mijn geheugen is ook niet meer wat het ooit geweest is....

Misschien even bespreken in een (pre-)refinement. De oplossing, bedoel ik, niet mijn geheugen. ;-)

**Comment by sergei-maertens** [comment]
klopt, hardcoded maken is makkelijkste optie hier! dat was mijn voorstel ;)

**Comment by MarcoKlerks** [comment]
Voor de refinement: ik heb de acceptatie criteria flink uitgebreid. Controleer o.a. even de grootte van de story.

**Comment by nijmra** [comment]
> Voor de refinement: ik heb de acceptatie criteria flink uitgebreid. Controleer o.a. even de grootte van de story.

@MarcoKlerks @sergei-maertens in de story wordt gesteld dat er één of meer document handelingen per document opgeslagen moeten kunnen worden. In het (niet heel recente) [data model](https://github.com/GeneriekPublicatiePlatformWoo/registratie-component/issues/2#issuecomment-2326140879) zie ik maar ruimte voor een enkele documenthandeling:

`documenthandeling:
      type: object
      properties:
        soortHandeling:
          type: string
          enum:  # uit waardelijst
            - foo
            - bar
        tijdstip:
          type: string
          format: date-time`

Dit moeten er dus meerdere worden(?)

**Comment by MarcoKlerks** [comment]
De DiWoo-standaard ondersteunt dit. Ik denk dat het in de praktijk weinig voorkomt. 

Het is prima om te beginnen met één handeling per document, maar het kan dus gebeuren dat gemeenten vragen om conform de standaard meer dan een handelingen per document te ondersteunen.

**Comment by MarcoKlerks** [comment]
Voorlopig vind ik https://github.com/GPP-Woo/GPP-burgerportaal/issues/157 een prima oplosisng.
